sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -161,140 +152,52 @@ AccessToken TokenCredentialImpl::ParseToken(
-
-bool GetPropertyValueAsInt64(
-    std::string const& jsonString,
-    std::string const& propertyName,
-    int64_t& outValue)
-{
-  auto const valueStartPos = GetPropertyValueStart(jsonString, propertyName);
-  if (valueStartPos == std::string::npos)
-  {
-    return false;
-  }
+    AccessToken accessToken;
+    accessToken.Token = parsedJson[accessTokenPropertyName].get<std::string>();
 
-  int64_t value = 0;
-  {
-    auto const size = jsonString.size();
-    for (auto pos = valueStartPos; pos < size; ++pos)
+    if (parsedJson.contains(expiresInPropertyName))
     {
-      auto c = jsonString[pos];
-      if (c < '0' || c > '9')
-      {
-        break;
-      }
-
-      value = (value * 10) + (static_cast<int64_t>(c) - '0');
+      accessToken.ExpiresOn
+          = std::chrono::system_clock::now()
+          + std::chrono::seconds(
+                parsedJson[expiresInPropertyName].get<std::chrono::seconds::duration::rep>());
     }
-  }
-
-  outValue = value;
-
-  return true;
-}
-
-std::string::size_type GetPropertyValueEnd(std::string const& str, std::string::size_type startPos);
-
-bool GetPropertyValueAsString(
-    std::string const& jsonString,
-    std::string const& propertyName,
-    std::string& outValue)
-{
-  auto const valueStartPos = GetPropertyValueStart(jsonString, propertyName);
-  if (valueStartPos == std::string::npos)
-  {
-    return false;
-  }
-  auto const jsonStringBegin = jsonString.begin();
-  outValue = std::string(
-      jsonStringBegin + valueStartPos,
-      jsonStringBegin + GetPropertyValueEnd(jsonString, valueStartPos));
-
-  return true;
-}
-
-std::string::size_type GetPropertyValueStart(
-    std::string const& jsonString,
-    std::string const& propertyName)
-{
-  auto const propertyNameValueSeparator = jsonString.find(':', jsonString.find(propertyName));
-  if (propertyNameValueSeparator == std::string::npos)
-  {
-    return std::string::npos;
-  }
-
-  auto pos = propertyNameValueSeparator;
-  {
-    auto const jsonStringSize = jsonString.size();
-    for (; pos < jsonStringSize; ++pos)
+    else if (expiresOnPropertyName.empty())
     {
-      auto c = jsonString[pos];
-      if (c != ':' && c != ' ' && c != '\"' && c != '\'')
-      {
-        break;
-      }
+      ThrowMissingJsonPropertyError(expiresInPropertyName);
     }
-  }
-
-  return pos;
-}
-
-std::string::size_type GetPropertyValueEnd(std::string const& str, std::string::size_type startPos)
-{
-  auto pos = startPos;
-  {
-    auto const strSize = str.size();
-    for (; pos < strSize; ++pos)
+    else if (!parsedJson.contains(expiresOnPropertyName))
+    {
+      ThrowMissingJsonPropertyError(expiresOnPropertyName);
+    }
+    else
     {
-      auto c = str[pos];
-      if (c == '\"' || c == '\'')
+      auto expiresOn = parsedJson[expiresOnPropertyName].get<std::string>();
       {
-        break;
+        auto const spacePos = expiresOn.find(' ');
+        if (spacePos != std::string::npos) 
```

Comment:
I wish this custom logic wasn't needed to modify the input JSON before turning it into our DateTime.
Why do we need to do this? Do we need a different parse method?

Reply:
So far, it is the only known input that has space in it instead of "T". If we have more, we can add (internally first, I think) another parse format support. It is easy to modify the parser to allow space there.
But, I would check with Jeffrey first to ask if he thinks it is safe to update the parser, and whether that still can be called `DateFormat::Rfc3339`.

RFC 3339 itself says:
"_NOTE: ISO 8601 defines date and time separated by "T". Applications using this syntax may choose, for the sake of readability, to specify a full-date and full-time separated by (say) a space character._"
https://www.rfc-editor.org/rfc/rfc3339

Reply:
#4171 4171